src/rcc.rs

```diff
@@ -45,92 +44,58 @@ impl RccExt for RCC {
 /// let mut rcc = dp.RCC.constrain();
 /// ```
 pub struct Rcc {
-    /// AMBA High-performance Bus (AHB) registers
-    pub ahb: AHB,
-    /// Advanced Peripheral Bus 1 (APB1) registers
-    pub apb1: APB1,
-    /// Advanced Peripheral Bus 2 (APB2) registers
-    pub apb2: APB2,
     pub cfgr: CFGR,
     pub bkp: BKP,
 }
 
 /// AMBA High-performance Bus (AHB) registers
-///
-/// Aquired through the `Rcc` registers:
-///
-/// ```rust
-/// let dp = pac::Peripherals::take().unwrap();
-/// let mut rcc = dp.RCC.constrain();
-/// function_that_uses_ahb(&mut rcc.ahb)
-/// ```
 pub struct AHB {
     _0: (),
 }
 
 impl AHB {
-    // TODO remove `allow`
-    #[allow(dead_code)]
-    pub(crate) fn enr(&mut self) -> &rcc::AHBENR {
-        // NOTE(unsafe) this proxy grants exclusive access to this register
-        unsafe { &(*RCC::ptr()).ahbenr }
+    fn enr(rcc: &RccRB) -> &rcc::AHBENR {
+        &rcc.ahbenr
     }
```

Comment:
Are these functions required anmyore, would it not be neater to just do `rcc.ahbenr` directly when we do need them?

Reply:
These functions are used in `bus!` macro.

Reply:
Ah right, good point